microceph/cmd/microceph/disk_list.go

```diff
@@ -71,34 +78,66 @@ func (c *cmdDiskList) Run(cmd *cobra.Command, args []string) error {
 }
 
 func listLocalDisks(cli *microCli.Client) error {
+	data := [][]string{}
+
+	disks := get_unpartitioned_disks(cli)
+	for _, disk := range disks {
+		data = append(data, []string{disk.Model, disk.Size, disk.Type, disk.Path})
+	}
+
+	fmt.Println("")
+	fmt.Println("Available unpartitioned disks on this system:")
+
+	header := []string{"MODEL", "CAPACITY", "TYPE", "PATH"}
+	sort.Sort(lxdCmd.SortColumnsNaturally(data))
+
+	err := lxdCmd.RenderTable(lxdCmd.TableFormatTable, header, data, data)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func get_unpartitioned_disks(cli *microCli.Client) []Disk {
 	// List configured disks.
 	disks, err := client.GetDisks(context.Background(), cli)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	// List physical disks.
 	resources, err := client.GetResources(context.Background(), cli)
 	if err != nil {
-		return err
+		return nil
 	}
 
 	// Get local hostname.
 	hostname, err := os.Hostname()
 	if err != nil {
-		return err
+		return nil
 	}
 
 	// Prepare the table.
-	data := [][]string{}
+	data := []Disk{}
 	for _, disk := range resources.Disks {
 		if len(disk.Partitions) > 0 {
 			continue
 		}
 
+		if len(disk.DeviceID) == 0 {
+			continue
+		}
+
+		// Minimum size set to 2GB i.e. 2*1024*1024*1024
+		if disk.Size < uint64(2147483648) {
```

Comment:
Making this a constant would probably help in future refactoring of the code.

Reply:
I would also at least log why some entries were omitted.

Reply:
ack